tools/calisp/calisp.xml

```diff
@@ -0,0 +1,298 @@
+<tool id="calisp" name="calisp" version="@TOOL_VERSION@+galaxy@VERSION_SUFFIX@" profile="21.05">
+    <description>Estimate isotopic composition of peptides from proteomics mass spectrometry data</description>
+    <macros>
+        <token name="@TOOL_VERSION@">3.0.10</token>
+        <token name="@VERSION_SUFFIX@">0</token>
+        <xml name="input_macro" tokens="multiple">
+            <!-- According to readme mzid input is not yet implented -->
+        </xml>
+    </macros>
+    <requirements>
+        <requirement type="package" version="@TOOL_VERSION@">calisp</requirement>
+    </requirements>
+    <command detect_errors="aggressive"><![CDATA[
+#import re
+
+mkdir -p spectra &&
+#if $spectrum_cond.spectrum_sel == "batch"
+    #set escaped_specs = re.sub('[^\w\-\.,:]', '_', str($spectrum_cond.spectrum_file.element_identifier))
+    ln -s '$spectrum_file' spectra/'$escaped_specs' &&
+#else
+    #for $spec in $spectrum_cond.spectrum_file
+        #set escaped_specs = re.sub('[^\w\-\.,:]', '_', str($spec.element_identifier))
+        ln -s '$spec' spectra/'$escaped_specs' &&
+    #end for
+#end if
+
+mkdir -p psms &&
+#if $peptide_cond.peptide_sel == "batch"
+    #set escaped_peps = re.sub('[^\w\-\.,:]', '_', str($peptide_cond.peptide_file.element_identifier))
+    ln -s '$peptide_file' psms/'$escaped_peps' &&
+#else
+    #for $pep in $peptide_cond.peptide_file
+        #set escaped_peps = re.sub('[^\w\-\.,:]', '_', str($pep.element_identifier))
+        ln -s '$pep' psms/'$escaped_peps' &&
+    #end for
+#end if
+
+calisp 
+    --spectrum_file spectra/
+    --peptide_file psms/
+    --output_file calisp-output/
+    --mass_accuracy $mass_accuracy
+    --bin_delimiter '$bin_delimiter'
+    --threads "\${GALAXY_SLOTS:-1}"
+    --isotope $isotope
+    $compute_clumps &&
+'$__tool_directory__/feather2tsv.py' --calisp_output calisp-output/
+    ]]></command>
+    <inputs>
+        <conditional name="spectrum_cond">
+            <param name="spectrum_sel" type="select" label="Spectra batch mode">
+                <option value="batch">Batch</option>
+                <option value="joint">Joint</option>
+            </param>
+            <when value="batch">
+                <param argument="--spectrum_file" type="data" multiple="false" format="mzml" label="Spectrum file"/>
+            </when>
+            <when value="joint">
+                <param argument="--spectrum_file" type="data" multiple="true" format="mzml" label="Spectrum files"/>
+            </when>
+        </conditional>
+        <conditional name="peptide_cond">
+            <param name="peptide_sel" type="select" label="Peptide batch mode">
```

Comment:
mh, batch-batch, batch-joint, joint-batch, joint-joint are all possible?

Reply:
Discussed with our users. Lets go with batch-batch only for now and lets see if other modes are demanded.